test/common_test/simulation_features.cc

```diff
@@ -0,0 +1,562 @@
+ * limitations under the License.
+ *
+*/
+#include <gtest/gtest.h>
+
+#include <set>
+#include <string>
+#include <unordered_set>
+
+#include <gz/common/Console.hh>
+
+#include <gz/math/eigen3/Conversions.hh>
+
+#include "../helpers/TestLibLoader.hh"
+
+#include <gz/physics/sdf/ConstructJoint.hh>
+#include <gz/physics/sdf/ConstructLink.hh>
+#include <gz/physics/sdf/ConstructModel.hh>
+#include <gz/physics/sdf/ConstructCollision.hh>
+#include <gz/physics/sdf/ConstructWorld.hh>
+
+#include "gz/physics/BoxShape.hh"
+#include <gz/physics/GetContacts.hh>
+#include "gz/physics/CylinderShape.hh"
+#include "gz/physics/CapsuleShape.hh"
+#include "gz/physics/EllipsoidShape.hh"
+#include <gz/physics/FreeGroup.hh>
+#include <gz/physics/GetBoundingBox.hh>
+#include "gz/physics/SphereShape.hh"
+
+#include <gz/physics/ConstructEmpty.hh>
+#include <gz/physics/FindFeatures.hh>
+#include <gz/physics/ForwardStep.hh>
+#include <gz/physics/GetEntities.hh>
+#include <gz/physics/RequestEngine.hh>
+
+#include <sdf/Root.hh>
+
+// The features that an engine must have to be loaded by this loader.
+using Features = gz::physics::FeatureList<
+  gz::physics::ConstructEmptyWorldFeature,
+
+  gz::physics::FindFreeGroupFeature,
+  gz::physics::SetFreeGroupWorldPose,
+  gz::physics::SetFreeGroupWorldVelocity,
+
+  gz::physics::GetContactsFromLastStepFeature,
+  gz::physics::CollisionFilterMaskFeature,
+
+  gz::physics::GetModelFromWorld,
+  gz::physics::GetLinkFromModel,
+  gz::physics::GetShapeFromLink,
+  gz::physics::GetModelBoundingBox,
+
+  // gz::physics::sdf::ConstructSdfJoint,
+  gz::physics::sdf::ConstructSdfLink,
+  gz::physics::sdf::ConstructSdfModel,
+  gz::physics::sdf::ConstructSdfCollision,
+  gz::physics::sdf::ConstructSdfWorld,
+
+  gz::physics::ForwardStep,
+
+  gz::physics::AttachBoxShapeFeature,
+  gz::physics::AttachSphereShapeFeature,
+  gz::physics::AttachCylinderShapeFeature,
+  gz::physics::AttachEllipsoidShapeFeature,
+  gz::physics::AttachCapsuleShapeFeature,
+  gz::physics::GetSphereShapeProperties,
+  gz::physics::GetBoxShapeProperties,
+  gz::physics::GetCylinderShapeProperties,
+  gz::physics::GetCapsuleShapeProperties,
+  gz::physics::GetEllipsoidShapeProperties
+>;
+
+using TestWorldPtr = gz::physics::World3dPtr<Features>;
+using TestContactPoint = gz::physics::World3d<Features>::ContactPoint;
+
+class SimulationFeaturesTest:
+public gz::physics::TestLibLoader
+{
+  // Documentation inherited
+  public: void SetUp() override
+  {
+    gz::common::Console::SetVerbosity(4);
+
+    loader.LoadLib(SimulationFeaturesTest::GetLibToTest());
+
+    // TODO(ahcorde): We should also run the 3f, 2d, and 2f variants of
+    // FindFeatures
+    pluginNames = gz::physics::FindFeatures3d<Features>::From(loader);
+    if (pluginNames.empty())
+    {
+      std::cerr << "No plugins with required features found in "
+                << GetLibToTest() << std::endl;
+      // TODO(ahcorde): If we update gtest we can use here GTEST_SKIP()
```

Comment:
Is GTEST_SKIP not available in the current version?  I can prioritize getting gtest updated.

Reply:
yep, it's not available